### PR TITLE
WIP: feat: add postSaveAction

### PR DESCRIPTION
**Pull request by mzndako**

### CODE REVIEW

File: sqfentity/lib/sqfentity.dart [hunk 1]
```diff
@@ -375,12 +385,27 @@ class SqfEntityProvider extends SqfEntityModelBase {
...
+
+          final keys = closeBracket.split(',');
+
+          final record = {};
+
+          for (int i = 0; i < params!.length; i++) {
+            record[keys[i]] = params[i];
+          }
+
+          _dbModel!.postSaveAction!(_tableName!, record, 'INSERT');
```
Comment (mzndako) [inline sqfentity/lib/sqfentity.dart]: we need to wait for `postSaveAction` to complete before ending the save. 
```suggestion
         await _dbModel!.postSaveAction!(_tableName!, record, 'INSERT');
```

Comment (mzndako) [reply sqfentity/lib/sqfentity.dart]: This `rawInsert` method is also called on update. The `pSql` string can also contain `OR REPLACE` if it's an update query. 

We can use this to determine if the post action is an `INSERT` OR `UPDATE`.  Lets use `INSERT` if its an insert and `UPSERT` if its an update

```suggestion
          await _dbModel!.postSaveAction!(_tableName!, record, pSql.include('OR REPLACE') ? 'UPSERT' : 'INSERT');
```

File: sqfentity/lib/sqfentity.dart [hunk 2]
```diff
@@ -292,6 +292,12 @@ class SqfEntityProvider extends SqfEntityModelBase {
     try {
       /// Leave it in this format for Throw to stay in this catch
       final res = await updateOrThrow(obj);
+
+      if (_dbModel!.postSaveAction != null) {
+        final record = obj.toMap(forQuery: true);
+        await _dbModel!.postSaveAction!(_tableName!, record, 'INSERT');
```
Comment (mzndako) [inline sqfentity/lib/sqfentity.dart]: This should be update 
```suggestion
        await _dbModel!.postSaveAction!(_tableName!, record, 'UPDATE');

```

Comment (bynalab) [reply sqfentity/lib/sqfentity.dart]: Haha, copy and paste mistake. Resolved.

**Comment by mzndako** [comment]
@bynalab 
Let's handle the following for this:
* add a new property to soft delete to use deletedAt for soft delete
* add preSaveAction for rawInsert as well

**Comment by mzndako** [comment]
This is no longer needed

**Pull request closed without merging**